src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+        Storage.people.clear();
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User validUser = new User();
+        validUser.setLogin("validUser123");
+        validUser.setPassword("qwertyQ123");
+        validUser.setAge(20);
+
+        User registeredUser = registrationService.register(validUser);
+        assertNotNull(registeredUser, "User should be registered");
+        assertEquals(validUser.getLogin(), registeredUser.getLogin());
+        assertEquals(1, Storage.people.size(), "User should be added to the storage");
+    }
+
+    @Test
+    public void register_nullUser_notOk() {
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(null));
+        assertEquals("User cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_loginTooShort_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_passwordTooShort_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("12345");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_ageBelow18_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(17);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("User must be at least 18 years old", thrown.getMessage());
```

Comment:
The test 'register_ageBelow18_notOk' is correct, but the checklist suggests adding tests for a user exactly at 18 and over 18 years old to cover all edge cases.